tests/appsec/iast/_test_iast_fixtures.py

```diff
@@ -225,14 +225,14 @@ def validate_request_reported(self, request, source_type=None):
         sources = iast["sources"]
         assert sources, "No source reported"
         if source_type:
-            assert source_type in {s.get("origin") for s in sources}
-            sources = [s for s in sources if s["origin"] == source_type]
+            assert any(x in self.source_type for x in {s.get("origin") for s in sources})
```

Comment:
`source_type` can be overwritten

```suggestion
            assert any(x in source_type for x in {s.get("origin") for s in sources})
```